pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4850,54 +4851,51 @@ cpdef to_offset(freq, bint is_period=False):
-                if (not is_period and
-                        name != name.upper() and
-                        name.lower() not in {"s", "ms", "us", "ns"} and
-                        name.upper().split("-")[0].endswith(("S", "E"))):
-                    warnings.warn(
-                        f"\'{name}\' is deprecated and will be removed "
-                        f"in a future version, please use "
-                        f"\'{name.upper()}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = name.upper()
-                if is_period and name.upper() in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.upper().startswith("Y"):
+                if not is_period:
+                    if name.upper() in c_OFFSET_REMOVED_FREQSTR:
                         raise ValueError(
-                            f"for Period, please use \'Y{name.upper()[2:]}\' "
-                            f"instead of \'{name}\'"
+                            f"\'{name}\' is no longer supported for offsets. Please "
+                            f"use \'{c_OFFSET_REMOVED_FREQSTR.get(name.upper())}\' "
+                            f"instead."
                         )
+                    if (name.upper() != name and
+                            name.lower() not in {"W-TUE", "s", "ms", "us", "ns"} and
+                            name.upper().split("-")[0].endswith(("S", "E"))):
+                        raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
+                if (is_period and
+                        name.upper() in c_OFFSET_TO_PERIOD_FREQSTR and
+                        name != "ms" and
+                        name.upper().split("-")[0].endswith(("S", "E"))):
                     if (name.upper().startswith("B") or
                             name.upper().startswith("S") or
                             name.upper().startswith("C")):
                         raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
                     else:
+                        name_msg = "".join(name.upper().split("E", 1))
                         raise ValueError(
-                            f"for Period, please use "
-                            f"\'{c_REVERSE_OFFSET_DEPR_FREQSTR.get(name.upper())}\' "
+                            f"for Period, please use \'{name_msg}\' "
                             f"instead of \'{name}\'"
                         )
-                elif is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
-                    if name.upper() != name:
+                if is_period:
+                    if name.upper() in c_OFFSET_REMOVED_FREQSTR:
+                        if name.upper() != name:
+                            raise ValueError(
+                                f"\'{name}\' is no longer supported, "
+                                f"please use \'{name.upper()}\' instead.",
+                            )
```

Comment:
is this right? if it's a period alias, then why is it in `c_OFFSET_REMOVED_FREQSTR`? can we have as clean a separation as possible now that we're deprecating things?